src/tests/ftest/launch.py

```diff
@@ -758,6 +895,11 @@ def replace_yaml_file(yaml_file, args, yaml_dir):
                         "  - Replacement: {} -> {}".format(
                             value, replacements[value]))
 
+    # if VMD pci address in present under nvme_data,
+    # set DAOS_DISABLE_VMD to False
+    if vmd_flag is True:
+        os.environ["DAOS_DISABLE_VMD"] = "False"
```

Comment:
This should only be set when:

- a bdev_list placeholder entry was replaced
- the replacement value was a VMD device

Reply:
Moved the if condtion after if replacements: